ibis/expr/operations/generic.py

```diff
@@ -477,3 +478,13 @@ def root_tables(self):
     def output_dtype(self):
         exprs = [*self.results, self.default]
         return rlz.highest_precedence_dtype(exprs)
+
+
+class _Negatable(metaclass=abc.ABCMeta):
+    @property
+    @abc.abstractmethod
+    def _negated_class(self) -> type:  # pragma: no cover
```

Comment:
I'd prefer implementing the `negate()` abstractmethod directly in childs to prevent confusion of `_negated_class`  classprop with the argument and return type annotation.

Reply:
Cool, done.